### PR TITLE
revamp sam docs + add vs code launch config

### PR DESCRIPTION
**Pull request by HarshCasper**
Fixes https://github.com/localstack/docs/issues/657

### CODE REVIEW

File: content/en/user-guide/integrations/aws-sam/index.md [hunk 1]
```diff
@@ -5,35 +5,87 @@ description: >
...
+                    "MAIN_DOCKER_NETWORK": "localstack-network"
+                }
+            },
+            "sam": {
+                "dockerNetwork": "localstack-network"
+            }
+        }
+```
+
+The `dockerNetwork` property is essential as it allows the LocalStack container to use the `samlocal invoke` commands within the same network as the LocalStack container itself. Adjust the Lambda function handler and environment variables as needed.
```
Comment (lakkeger) [inline content/en/user-guide/integrations/aws-sam/index.md:91]: The mentioned issue (#1234 ) is about `sam local` and NOT `samlocal`. By default the container is span up in the main docker network and accessible, however IF the user decides to run LocalStack on a different network ONLY then they need to modify this.

File: content/en/user-guide/integrations/aws-sam/index.md [hunk 2]
```diff
@@ -5,35 +5,87 @@ description: >
...
+### Install the `samlocal` wrapper script
+
+You can install the `samlocal` wrapper script by running the following command:
 
 {{< command >}}
 $ pip install aws-sam-cli-local
 {{< / command >}}
 
-### Usage
+### Create a new Pulumi project
```
Comment (lakkeger) [inline content/en/user-guide/integrations/aws-sam/index.md:27]:
```suggestion
### Create a new SAM project
```
It's not Pulumi :)

**Review by lakkeger — CHANGES_REQUESTED** [review]

File: content/en/user-guide/integrations/aws-sam/index.md (hunk 1 above)
Comment (MarcelStranak) [reply content/en/user-guide/integrations/aws-sam/index.md:91]: Here is the communication from the reporter, [channel](https://localstack-community.slack.com/archives/C03F7EQM2M8/p1686305656429799).

Comment (lakkeger) [reply content/en/user-guide/integrations/aws-sam/index.md:91]: He luckily shared the actual LS startup command and it seems he puts his LS instance deliberately to a different network.
image
Regarding the question if he uses `samlocal local` or `sam local` is unclear, but doesn't really matter as he's setting the env vars for the debug session in the shared config.

**Review by MarcelStranak — CHANGES_REQUESTED** [review]

File: content/en/user-guide/integrations/aws-sam/index.md [hunk 3]
```diff
@@ -5,35 +5,87 @@ description: >
...
+Enter the default values for the deployment, such as the stack name, region, and confirm the changes. The `samlocal` wrapper will package and deploy the application to LocalStack.
 
 ### Configuration
 
-* `AWS_ENDPOINT_URL`: The endpoint URL (i.e., protocol, host, and port) to connect to LocalStack (default: `http://localhost:4566`)
+| Environment Variable   | Default value                                    | Description                                                             |
+|------------------------|--------------------------------------------------|-------------------------------------------------------------------------|
+| AWS_ENDPOINT_URL       | `http://localhost.localstack.cloud:4566`        | URL at which the `boto3` client can reach LocalStack                   |
+| EDGE_PORT              | `4566` (Deprecated)                              | Port number under which the LocalStack edge service is available        |
+| LOCALSTACK_HOSTNAME    | `localhost` (Deprecated)                         | Host under which the LocalStack edge service is available
```
[the remark this answers is not in the document]
Comment (lakkeger) [reply content/en/user-guide/integrations/aws-sam/index.md:55]: I'd keep them until we provide a new version of the wrapper.

**Pull request merged**